### ISSUE TITLE
Not getting 48% training accuracy in one epoch

### ISSUE THREAD

**Issue by lorenzogatto** [srebuffi/iCaRL#1]
Hello,

I'm trying to execute the code in this repository as suggested in the **Nota Bene** section of the readme, that is, changing the number of epochs to 1 and get the accuracy for the first two batches.
I'm only getting **24% accuracy** for the first batch (readme says it should be 48%), while I'm getting the expected 20% accuracy for the second batch.
I did not change any of the code except the three folders.
To make sure we are using the same data, my training set is made of 1281167 images.
I wonder if I should change the learning rate, if the reported 48% is wrong, or if there is some other issue.
I attach the partial output, any help would be appreciated!
https://pastebin.com/k2PvAWz3

Thanks

**Comment by srebuffi** [comment]
Hello,

I thank you for your feedback. I think it comes from the settings: I might have gotten this with total_classes = 100 and classes_per_group = 10. I will dig into it after the incoming CVPR deadline (Nov. 15th).

Thanks again,

**Comment by srebuffi** [comment]
Hi,
Did you try with total_classes=100 and classes_per_group=10 ?

**Comment by lorenzogatto** [comment]
Hi,

I can confirm that reducing the classes per group to 10, an accuracy of 48% can be obtained in one epoch.
Moreover, I noticed that changing with tf.device('/cpu:0'): to with tf.device('/gpu:0'): in main_resnet_tf.py increases code speed by an order of magnitude, so the tests now run much faster :)

Thanks!

**Comment by srebuffi** [comment]
Great! Thanks for pointing it out. I will set it to  tf.device('/gpu:0'): . 

Thanks for the feedback!
